functions/__ssh_agent_is_started.fish

```diff
@@ -1,9 +1,18 @@
 function __ssh_agent_is_started -d "check if ssh agent is already started"
-	if begin; test -f $SSH_ENV; and test -z "$SSH_AGENT_PID"; end
+	if test -n "$SSH_CONNECTION"
+		# This is an SSH session
+		ssh-add -l > /dev/null 2>&1
+		if test $status -eq 0 -o $status -eq 1
```

Comment:
Q: Why did you check if status was 1 here? The man page says:

> EXIT STATUS
>     Exit status is 0 on success, 1 if the specified command fails, and 2 if ssh-add is unable to contact the authentication agent.

So if it returns 1, should that not be considered a failure i.e. t hat the agent is not properly forwarded?

Reply:
Example:

```fish
    if test -n "$SSH_CONNECTION"
        # This is an SSH session
        set -l output (ssh-add -l 2>&1)
        if test $status -eq 0
            # An SSH agent was forwarded
            return 0
        else if test $status -eq 1
            if test "$output" = "The agent has no identities."
                # An SSH agent was forwarded but has no identities
                return 0
            end
        end
    end
```